index.php

```diff
@@ -46,8 +46,46 @@
 define('NO_UPGRADE_CHECK', true);
 define('ABORT_AFTER_CONFIG', true);
 
-require_once('../../../config.php');
-global $CFG;
+$configfile = '../../../config.php';
+
+/**
+ * Checks if the command line maintenance mode has been enabled. Skip the config bootstrapping.
+ *
+ * @param string $configfile The relative path for config.php
+ * @return bool|string False if not found, or the $CFG->dataroot path if found.
+ */
+function checkclimaintenance($configfile) {
+    $content = file_get_contents($configfile);
+    $content = preg_replace("/\/\//", "\n//", $content);  // Set comments to be on newlines, replace '//' with '\n//'
+    $content = preg_replace("/;/", ";\n", $content);      // Split up statements, replace ';' with ';\n'
+    $content = preg_replace("/^[\s]+/m", "", $content);   // Removes all initial whitespace and newlines.
+
+    $lines = explode(PHP_EOL, $content);
+    foreach ($lines as $line) {
+        $line = trim($line);
+        $re = '/^\$CFG->dataroot\s+=\s+["\'](.*)["\'];/';  // Lines starting with $CFG->dataroot
```

Comment:
Can you please test this with this which I think will break it:

```$CFG->dataroot = "/bar"; # this used to be "/foo";```

Try changing ```(.*)``` to non-greedy ```(.*?)```